src/main/java/net/datafaker/service/WeightedRandomSelector.java

```diff
@@ -0,0 +1,126 @@
+package net.datafaker.service;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+
+/**
+ * A utility class for selecting a random element from a list based on assigned weights.
+ **/
+public class WeightedRandomSelector {
+    private static final String WEIGHT_KEY = "weight";
+    private static final String VALUE_KEY = "value";
+    private final Random random;
+
+    public WeightedRandomSelector(Random random) {
+        this.random = random != null ? random : new Random();
+    }
+
+    public Random getRandom() {
+        return random;
+    }
+
+    /**
+     * Returns a weighted random element from the given list, where each element is represented as a Map
```

Comment:
Am I right that the only way to change weight is change file and rebuild everything?

Reply:
The weights are provided as part of the input items list and are processed at runtime. Changing weights does not require rebuilding the application but simply involves modifying the input list or the file (if applicable) and rerunning the program.